asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
If you're concerned about the API surface, you don't have to export those methods either.  I made the conscious decision to export them in go-github, but you don't necessarily need to, since you just need them for internal use.  If you went that route, you'd probably want to refactor the existing `request` method to use the new `newRequest` and `do` methods, but it wouldn't change the API at all.  Just a thought.